plugins/org.obeonetwork.m2doc.sirius/src/org/obeonetwork/m2doc/sirius/SiriusDiagramByRepresentationAndEObjectProvider.java

```diff
@@ -0,0 +1,203 @@
+package org.obeonetwork.m2doc.sirius;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.NullProgressMonitor;
+import org.eclipse.core.runtime.Path;
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.emf.ecore.util.EcoreUtil;
+import org.eclipse.emf.edit.domain.EditingDomain;
+import org.eclipse.gmf.runtime.diagram.core.preferences.PreferencesHint;
+import org.eclipse.gmf.runtime.diagram.core.util.ViewUtil;
+import org.eclipse.gmf.runtime.diagram.ui.image.ImageFileFormat;
+import org.eclipse.gmf.runtime.diagram.ui.render.util.CopyToImageUtil;
+import org.eclipse.gmf.runtime.notation.Diagram;
+import org.eclipse.sirius.business.api.dialect.DialectManager;
+import org.eclipse.sirius.business.api.session.CustomDataConstants;
+import org.eclipse.sirius.business.api.session.Session;
+import org.eclipse.sirius.business.api.session.SessionManager;
+import org.eclipse.sirius.diagram.DDiagram;
+import org.eclipse.sirius.diagram.DSemanticDiagram;
+import org.eclipse.sirius.diagram.business.api.query.DDiagramQuery;
+import org.eclipse.sirius.diagram.business.internal.metamodel.spec.DSemanticDiagramSpec;
+import org.eclipse.sirius.diagram.description.DiagramDescription;
+import org.eclipse.sirius.diagram.ui.tools.api.part.DiagramEditPartService;
+import org.eclipse.sirius.viewpoint.DRepresentation;
+import org.eclipse.sirius.viewpoint.DView;
+import org.eclipse.sirius.viewpoint.description.AnnotationEntry;
+import org.eclipse.sirius.viewpoint.description.Viewpoint;
+import org.obeonetwork.m2doc.provider.DiagramProvider;
+import org.obeonetwork.m2doc.provider.IProvider;
+import org.obeonetwork.m2doc.provider.OptionType;
+import org.obeonetwork.m2doc.provider.ProviderConstants;
+import org.obeonetwork.m2doc.provider.ProviderException;
+
+/**
+ * {@link SiriusDiagramByRepresentationAndEObjectProvider} are used to get sirius diagrams from an
+ * eObject and a name.
+ * 
+ * @author Romain Guider
+ */
+public class SiriusDiagramByRepresentationAndEObjectProvider extends DiagramProvider {
+	/**
+	 * The option key used to retrieve the {@link EObject} root of some diagram representations. 
+	 */
+	private static final String KEY_TARGET_ROOT_OBJECT = "rootObject";
+	/**
+	 * The key used in the map passed to {@link IProvider} to define the Sirius
+	 * representation title from which image will be computed.
+	 */
+	private static final String KEY_DIAGRAM_DESCRIPTION_NAME = "diagramDescriptionName";
```

Comment:
same remark as with the previous constant : I would name it DIAGRAM_DESCRIPTION_NAME_KEY